File: frontend/src/types/company.ts
export interface Company {
  id?: number;
  name: string;
  description?: string;
  contactPerson?: string;
  email?: string;
  phone?: string;
  address?: string;
  website?: string;
  taxId?: string;
  paymentTerms?: number;
  status?: string; // Assuming CompanyStatus enum is represented as a string
  createdAt?: string; // Assuming LocalDateTime is represented as a string
  updatedAt?: string; // Assuming LocalDateTime is represented as a string
}

export interface CompanyContact {
  id?: number;
  companyId: number; // Link to the Company
  name: string;
  email?: string;
  phone?: string;
  role?: string;
  isPrimary?: boolean;
  createdAt?: string; // Assuming LocalDateTime is represented as a string
} 

File: backend/company-service/src/main/java/com/myworkmanagement/company/entity/Company.java
package com.myworkmanagement.company.entity;

import com.myworkmanagement.company.config.CompanyStatusConverter;
import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "companies")
public class Company {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Company name is required")
    @Column(nullable = false, unique = true)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String contactPerson;

    @Email(message = "Invalid email format")
    @Pattern(regexp = "^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}$", message = "Invalid email format")
    @Column(unique = true)
    private String email;

    private String phone;

    @Column(columnDefinition = "TEXT")
    private String address;

    private String website;

    @Column(name = "tax_id", unique = true)
    private String taxId;

    @Column(name = "payment_terms")
    private Integer paymentTerms;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CompanyStatus status;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @Column(name = "user_email", nullable = false)
    private String userEmail;

} 